pkg/api/norman/customization/aks/handler.go

```diff
@@ -239,10 +247,9 @@ func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credI
 	if cap.AuthBaseURL == "" {
 		cap.AuthBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
 	}
-	region := req.URL.Query().Get("region")
-	if region != "" {
-		cap.ResourceLocation = region
-	}
+
+	cap.ResourceLocation = req.URL.Query().Get("region")
```

Comment:
What is the intention of this change? if cap has a value then this would blank it out if it's not in the request

Reply:
This was discussed here: https://github.com/rancher/rancher/pull/34624#discussion_r701542177

Reply:
Kinda? My question still stands, if cap has a value and the request doesn't we are going to blank it... then what?

Reply:
When I traced this myself it looked like it would not ever have a value as things currently are.

Reply:
The only place this unexported func is called passes empty `Capabilities` object, so immediate usage is not a concern as there is nothing that could be overwritten.
Overall, it looks to me that the goal of this internal func is to initialize capabilities from request, not to augment existing capabilities passed to it - judging by its current usage. So it would be more fitting for the func to return capabilities rather than take a pointer to one.
Given the current signature, I do see your point and don't mind reinstating this check. Feels like we're splitting hairs here though a bit.

Reply:
@paynejacob , I see this was not changed. Did you meet with Dan and agree not to address it?